File: testrandom.f90
program testrandom

real (kind=8) :: seed

interface
    real(kind=8) function random(seed)
      real (kind=8)  :: seed
    end function random
end interface


seed = 1.0D0

print *, seed
do i = 1, 10
  print *, random(seed)
enddo

end ! program testrandom

File: testrandom.c
#include <stdio.h>
#include <stdlib.h>

double random(double *seed);

int main(){

double seed;
int i;

seed = 1.000000000000000;

printf("%f\n", seed);
for (i=0; i<10; i++ ) {
    printf("%f\n", random(&seed));
}

}
